packages/block-library/package.json

```diff
@@ -57,6 +57,7 @@
 		"@wordpress/private-apis": "file:../private-apis",
 		"@wordpress/reusable-blocks": "file:../reusable-blocks",
 		"@wordpress/rich-text": "file:../rich-text",
+		"@wordpress/router": "file:../router",
```

Comment:
Router shouldn't be a dependency of the block library package as not every editor has one.

@glendaviesnz added a way to handle navigation in #57036 via an editor setting that will work across the post and site editors.

Reply:
Thanks for the pointer Dan 👍

Reply:
There may be some modifications coming to that approach now as we need to extend it a little probably in order to get rid of the template-only mode, will hopefully get that sorted today.

Reply:
> the template-only mode

What is this mode you speak of?

Reply:
> What is this mode you speak of?

https://github.com/WordPress/gutenberg/blob/trunk/packages/editor/src/store/actions.js#L579

It was added temporarily as part of the editor unification work, but we are now in the process of pulling it out again. Context being we wanted a way to edit a [synced pattern in isolation in the post editor](https://github.com/WordPress/gutenberg/pull/57036), so we looked at adding a `pattern-only` mode, but it was then decided that this was not very scalable as we didn't want modes for lots of different entities.

It was decided that a better approach would be to change the entity that was passed into the editor provider, and we added a [`getPostLinkProps`](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-post/src/editor.js#L36) editor setting which is a callback that returns an `onClick` event and `href` that changes the current entity in the editor.

We initially just used this for [editing synced patterns in the post editor (https://github.com/WordPress/gutenberg/pull/57036) but I am now looking at doing the same for [editing templates from the post editor](https://github.com/WordPress/gutenberg/pull/57700), which currently uses the `template-only mode. When working through this our initially implementation doesn't extend quite so  well for the wider use case, so I have [a suggested refactoring of this here](https://github.com/WordPress/gutenberg/pull/57745).